apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/QueryUtils.java

```diff
@@ -17,6 +19,47 @@
 
 /** As set of methods to help form JPA queries. */
 public class QueryUtils {
+  /**
+   * Database function that checks all claims for a given beneficiaryId and returns a bitwise mask
+   * value that shows if a given claim type will have any data.
+   */
+  public static final String CHECK_CLAIMS_FOR_DATA =
+      "SELECT * FROM check_claims_mask(:beneIdValue)";
+
+  /** BitSet index identifier for Carrier Claims. */
+  public static final int CARRIER_HAS_DATA = 0;
+  /** BitSet index identifier for Inpatient Claims. */
+  public static final int INPATIENT_HAS_DATA = 1;
+  /** BitSet index identifier for Outpatient Claims. */
+  public static final int OUTPATIENT_HAS_DATA = 2;
+  /** BitSet index identifier for SNF Claims. */
+  public static final int SNF_HAS_DATA = 3;
+  /** BitSet index identifier for DME Claims. */
+  public static final int DME_HAS_DATA = 4;
+  /** BitSet index identifier for HHA Claims. */
+  public static final int HHA_HAS_DATA = 5;
+  /** BitSet index identifier for Hospice Claims. */
+  public static final int HOSPICE_HAS_DATA = 6;
+  /** BitSet index identifier for Part D Events. */
+  public static final int PART_D_HAS_DATA = 7;
+
+  /** bitwise value denoting CARRIER_CLAIMS data for a beneficiary. */
+  private static final int V_CARRIER_HAS_DATA = (1 << 0);
+  /** bitwise value denoting INPATIENT_CLAIMS data for a beneficiary. */
+  private static final int V_INPATIENT_HAS_DATA = (1 << 1);
+  /** bitwise value denoting OUTPATIENT_CLAIMS data for a beneficiary. */
+  private static final int V_OUTPATIENT_HAS_DATA = (1 << 2);
+  /** bitwise value denoting SNF_CLAIMS data for a beneficiary. */
+  private static final int V_SNF_HAS_DATA = (1 << 3);
+  /** bitwise value denoting DME_CLAIMS data for a beneficiary. */
+  private static final int V_DME_HAS_DATA = (1 << 4);
+  /** bitwise value denoting HHA_CLAIMS data for a beneficiary. */
+  private static final int V_HHA_HAS_DATA = (1 << 5);
+  /** bitwise value denoting HOSPICE_CLAIMS data for a beneficiary. */
+  private static final int V_HOSPICE_HAS_DATA = (1 << 6);
+  /** bitwise value denoting PARTD_EVENTS data for a beneficiary. */
+  private static final int V_PART_D_HAS_DATA = (1 << 7);
+
```

Comment:
What do you think about doing something like this? It would eliminate the logic inside `QueryUtils` and make it so we don't need to manage so many constant values.

```
public static BitSet hasClaimsData(EntityManager entityManager, long bene_id) {
   boolean carrier_has_data, inpatient_has_data, outpatient_has_data; // etc..
   carrier_has_data = hospice_has_data = outpatient_has_data = false; // etc..
   boolean[] claim_has_data_by_bit_pos = { carrier_has_data, hospice_has_data, outpatient_has_data };

   List<Object> values =
      entityManager
         .createNativeQuery(CHECK_CLAIMS_FOR_DATA)
         .setParameter("beneIdValue", bene_id)
         .getResultList();
   Integer maskVal = (Integer) (values != null && values.size() > 0 ? values.get(0) : 0);
   BitSet rslt = new BitSet(maskVal);
   for (int bit_pos = 0; bit_pos < claim_has_data_by_bit_pos.length; bit_pos++) {
      rslt.set(bit_pos, (maskVal & (1 << bit_pos)) == 1);
   }
   return rslt;
}
```
then...
```
public boolean hasClaimsDataByType(ClaimTypeV2 claimType, BitSet bitset) {
   return switch (claimType) {
      case ClaimTypeV2.Carrier -> bitSet.get(0) == 1;
      case ClaimTypeV2.Inpatient -> bitSet.get(1) == 1;
      ...
      default -> false;
   }
```
both methods residing in EOB endpoint instead of `QueryUtils`

Reply:
So the Client[12).java is an implementation direction that should never have occurred. In fact, Karl ranted vs. that as a mistake in how the devs implemented this. In fact, those two classes are in line to become a single class if/when we have IOC in which we can inject the *Transformer class.

As for the other suggestion, not sure why we'd want to duplicate code in two EOBs vs a single QueryUtils serving both V1 and V2. As for the for loop, it's possible, but feels brittle in that it becomes more position-based vs. just decomposing the mask.

Reply:
Just noting here that `(maskVal & (1 << bit_pos)) == 1` is incorrect because the bitwise AND does not return 1 or zero.  It returns an integer that includes only those bits which are 1 in both integers.  So the above would need to be something like:
```
int bit = (1 << bit_pos);
boolean bitIsSet =(maskVal & bit) == bit;
```

Reply:
Aren't the variables you've declared already position-based though? I don't see how one `LinkedHashMap` is any more or less brittle than 16 variables, but I do understand your concern with code-duplication. I'm not sure if it's better to have dup code or if it's better to have EOB-exclusive functionality in a shared utility class. I guess my preference would be having the duplicated code but I understand that's my personal opinion.